vault/audited_headers.go

```diff
@@ -0,0 +1,116 @@
+package vault
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+const (
+	// Key used in the BarrierView to store and retrieve the header config
+	auditedHeadersEntry = "audited-headers"
+	// Path used to create a sub view off of BarrierView
+	auditedHeadersSubPath = "audited-headers-config/"
+)
+
+type auditedHeaderSettings struct {
+	HMAC bool `json:"hmac"`
+}
+
+// AuditedHeadersConfig is used by the Audit Broker to write only approved
+// headers to the audit logs. It uses a BarrierView to persist the settings.
+type AuditedHeadersConfig struct {
+	Headers map[string]*auditedHeaderSettings
```

Comment:
Can we rename this to `HeaderSettings` or something, to better indicate what this actually holds?